ui/raid/import_export.ts

```diff
@@ -142,301 +136,799 @@ class RaidWCLImporter extends Importer {
+			console.error("Could not find matching preset for non buff bot", {
+				"name": this.name,
+				"spec": this.spec,
+				"type": this.type,
+				"talents": this.talents,
+			});
+			return;
+		}
+		
+		player = withSpecProto(this.spec, player, matchingPreset.rotation, specFuncs.talentsCreate(), matchingPreset.specOptions);
+		
+		player.talentsString = matchingPreset.talents;
+		player.consumes = matchingPreset.consumes;
+		
+		player.name = this.name;
+		player.class = nameToClass(this.type);
+		player.equipment = this.getEquipment();
+		player.race = matchingPreset.defaultFactionRaces[this.faction];
+		
+		return player;
+	}
+	
+	public getBuffBot(): BuffBot | undefined {
+		
+		if (!this.isBuffBot) {
+			return;
+		}
+		
+		const botID = buffBotNames[this.specType];
+		
+		if (botID == null) {
+			console.error("Buff Bot Spec not implemented: ", this.specType);
+			return;
+		}
+		
+		const bot = BuffBot.create();
+		bot.id = botID;
+		bot.raidIndex = -1; // Default it for now. // numPlayers
+		
+		return bot;
+	}
+	
+	public getPartyAuraIds() {
+		
+		const allSpecClassAuras: any = {
+			"Paladin": [
+				19746, // Concentration Aura
+				27149, // Devotion Aura,
+				27150, // Retribution Aura
+			],
+			"Warrior": [
+				2048,  // Battle Shout
+				469, // Commanding Shout
+			],
+			"Warlock": [
+				27268, // Pet Imp: Blood Pact
+				18696, // Improved Imp: Blood Pact
+			],
+		};
+		
+		// Reused for the plethora of Feral specs.
+		const feralDruidSpecAuras = [
+			24932, // Improved Leader of the Pack // at least 0,32,0
+			// 17007, // Leader of the Pack // at least 0,31,0
+		];
+		
+		// TODO: Could additionally filter out buff IDs based on minimum req talent strings?
+		const specSpecificAuras: any = {
+			"RetributionPaladin": [
+				20092, // Improved Retribution Aura // at least 0,0,16
+				20218, // Sanctity Aura // at least 0,0,21
+				31870, // Improved Sanctity Aura // at least 0,0,22
+			],
+			"GuardianDruid": [...feralDruidSpecAuras],
+			"WardenDruid": [...feralDruidSpecAuras],
+			"FeralDruid": [...feralDruidSpecAuras],
+			"BalanceDruid": [
+				24907, // Moonkin Aura // at least 31,0,0
+			],
+			"RestorationDruid": [
+				34123, // Tree of Life // at least 0,0,41
+			],
+			"MarksmanHunter": [
+				27066, // Trueshot Aura // at least 0,32,0
+			],
+			"EnhancementShaman": [
+				30811, // Unleashed Rage // at least 0,36,0
+			],
+			// "ElementalShaman": [] // Totem buffs do not show up in logs. Leaving for future reference.
+		};
+		
+		const consumableAuras = [
+			351355, // Greater Drums of Battle
+		];
+		
+		const classAuras = allSpecClassAuras[this.type] ?? [];
+		const specAuras = specSpecificAuras[this.specType] ?? [];
+		
+		const reliableAuras = [
+			...specAuras, ...classAuras, ...consumableAuras,
+		];
+		
+		if (this.type === "Shaman") {
+			// Shamans get moved around a lot, so Heroism isn't a good reference for what group they are in.
+			return [
+				...reliableAuras,
+				32182, // Heroism
```

Comment:
so I don't fully understand why this is included if it isn't reliable?

Reply:
It's "unreliable" because Shamans get moved around, so because of that,it's one of the last checks, so If all of the previous ones fail, it'll run it, but exit early if another one hits first.

Reply:
But it doesn't need to be included if you don't think it's valuable.

Reply:
ok, i understand better now. I see now that it is the kind of "final fallback" kind of check. Works for me!